### DISCUSSION TITLE
Why are the .hpp files in include/bit7z folder instead of just include?

### DISCUSSION THREAD

**Discussion by ra-dave** [rikyoz/bit7z#143] [General]
Just curious.

**Comment by rikyoz** [comment]
This is actually quite a recent change in the project structure! Before, it used to have just the `include` folder.

I decided to add the `bit7z` subfolder to make the project follow a structure common in C/C++ projects, where the public headers are separated from the rest of the code and put in an `include/<project name>` path.

This allows you to find the public API quickly and makes it easier to individuate which library an included header belongs to since, when you use it, you have the library prefix, e.g. `#include <bit7z/bitarchivewriter.hpp>`.
The downside is, of course, that it is a bit more verbose to type, but usually, you do not have to type it often!
